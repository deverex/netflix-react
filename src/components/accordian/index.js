import React, { useContext, createContext, useState } from "react";
import {
  Container,
  Inner,
  Title,
  Item,
  Header,
  Body,
} from "./styles/accordian";

const ToggleContext = createContext();

const Accordian = ({ children, ...restProps }) => {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
};

const AccordianTitle = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const AccordianItem = ({ children, ...restProps }) => {
  const [toggleShow, setToggleShow] = useState(false);

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

const AccordianHeader = ({ children, ...restProps }) => {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);
  return (
    <Header
      onClick={() => setToggleShow((toggleShow) => !toggleShow)}
      {...restProps}
    >
      {children}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

const AccordianBody = ({ children, ...restProps }) => {
  const { toggleShow } = useContext(ToggleContext);
  return toggleShow ? <Body {...restProps}>{children}</Body> : null;
};

Accordian.Title = AccordianTitle;
Accordian.Item = AccordianItem;
Accordian.Header = AccordianHeader;
Accordian.Body = AccordianBody;

export default Accordian;
