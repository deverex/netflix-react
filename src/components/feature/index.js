import React from "react";
import { Container, Title, SubTitle } from "./styles/feature";

const Feature = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const FeatureTitle = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const FeatureSubTitle = ({ children, ...restProps }) => {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Feature.Title = FeatureTitle;
Feature.SubTitle = FeatureSubTitle;

export default Feature;
