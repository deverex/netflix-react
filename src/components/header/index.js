import React from "react";
import { Link as ReactRouterLink } from "react-router-dom";

import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  FeatureCallOut,
  Text,
  Link,
  Group,
  Picture,
  Profile,
  Dropdown,
} from "./styles/header";

const Header = ({ bg = true, children, ...restProps }) => {
  return bg ? <Background {...restProps}>{children}</Background> : children;
};

const HeaderPicture = ({ src, ...restProps }) => {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

const HeaderProfile = ({ children, ...restProps }) => {
  return <Profile {...restProps}>{children}</Profile>;
};

const HeaderDropdown = ({ children, ...restProps }) => {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

const HeaderFeature = ({ children, ...restProps }) => {
  return <Feature {...restProps}>{children}</Feature>;
};

const HeaderFeatureCallOut = ({ children, ...restProps }) => {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

const HeaderFrame = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const HeaderGroup = ({ children, ...restProps }) => {
  return <Group {...restProps}>{children}</Group>;
};

const HeaderLogo = ({ to, children, ...restProps }) => {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

const HeaderButtonLink = ({ children, ...restProps }) => {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

const HeaderText = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const HeaderTextLink = ({ children, ...restProps }) => {
  return <Link {...restProps}>{children}</Link>;
};

Header.Frame = HeaderFrame;
Header.Logo = HeaderLogo;
Header.ButtonLink = HeaderButtonLink;
Header.Feature = HeaderFeature;
Header.FeatureCallOut = HeaderFeatureCallOut;
Header.Text = HeaderText;
Header.TextLink = HeaderTextLink;
Header.Group = HeaderGroup;
Header.Picture = HeaderPicture;
Header.Profile = HeaderProfile;
Header.Dropdown = HeaderDropdown;

export default Header;
