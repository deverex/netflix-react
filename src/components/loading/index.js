import React from "react";
import { Spinner, LockBody, Picture, ReleaseBody } from "./styles/loading";

const Loading = ({ src, ...restProps }) => {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} />
    </Spinner>
  );
};

const LoadingReleaseBody = () => {
  return <ReleaseBody />;
};

Loading.ReleaseBody = LoadingReleaseBody;

export default Loading;
