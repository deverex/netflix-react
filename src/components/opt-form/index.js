import React from "react";
import { Container, Input, Button, Text, Break } from "./styles/opt-form";

const OptForm = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const OptFormInput = ({ ...restProps }) => {
  return <Input {...restProps} />;
};

const OptFormButton = ({ children, ...restProps }) => {
  return (
    <Button {...restProps}>
      {children}
      <img src="/images/icons/chevron-right.png" alt="Try Now" />
    </Button>
  );
};

const OptFormText = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const OptFormBreak = ({ children, ...restProps }) => {
  return <Break {...restProps} />;
};

OptForm.Button = OptFormButton;
OptForm.Input = OptFormInput;
OptForm.Text = OptFormText;
OptForm.Break = OptFormBreak;

export default OptForm;
