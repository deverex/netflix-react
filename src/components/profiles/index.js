import React from "react";
import { Container, Title, Item, List, Picture, Name } from "./styles/profiles";

const Profiles = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const ProfilesTitle = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const ProfilesList = ({ children, ...restProps }) => {
  return <List {...restProps}>{children}</List>;
};

const ProfilesItem = ({ children, ...restProps }) => {
  return <Item {...restProps}>{children}</Item>;
};

const ProfilesUser = ({ children, ...restProps }) => {
  return <Item {...restProps}>{children}</Item>;
};

const ProfilesPicture = ({ children, src, ...restProps }) => {
  return (
    <Picture
      {...restProps}
      src={src ? `/images/users/${src}.png` : "/images/misc/loading.gif"}
    />
  );
};

const ProfilesName = ({ children, ...restProps }) => {
  return <Name {...restProps}>{children}</Name>;
};

Profiles.Title = ProfilesTitle;
Profiles.List = ProfilesList;
Profiles.Item = ProfilesItem;
Profiles.User = ProfilesUser;
Profiles.Picture = ProfilesPicture;
Profiles.Name = ProfilesName;

export default Profiles;
