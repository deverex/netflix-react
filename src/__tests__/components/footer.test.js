import React from "react";
import { render } from "@testing-library/react";
import { Footer } from "../../components";

describe("<Footer />", () => {
  it("renders the <Footer /> with populated data", () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions ? Contact Us.</Footer.Title>
        <Footer.Break></Footer.Break>
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Investor Relations</Footer.Link>
            <Footer.Link href="#">Ways to Watch</Footer.Link>
            <Footer.Link href="#">Corporate Information</Footer.Link>
            <Footer.Link href="#">Netflix Originals</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Help Centre</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact Us</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link htef="#">Account</Footer.Link>
            <Footer.Link htef="#">Redeem Gift Card</Footer.Link>
            <Footer.Link htef="#">Privacy</Footer.Link>
            <Footer.Link htef="#">Speed Test</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link htef="#">Buy Gift Cards</Footer.Link>
            <Footer.Link htef="#">Cookie Preferences</Footer.Link>
            <Footer.Link htef="#">Legal Notices</Footer.Link>
            <Footer.Link htef="#">Corporate Information</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Break></Footer.Break>
        <Footer.Text>Netflix United States</Footer.Text>
      </Footer>
    );

    expect(getByText("Questions ? Contact Us.")).toBeTruthy();
    expect(getByText("FAQ")).toBeTruthy();
    expect(getByText("Investor Relations")).toBeTruthy();
    expect(getByText("Ways to Watch")).toBeTruthy();
    expect(getByText("Account")).toBeTruthy();
    expect(getByText("Netflix Originals")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
