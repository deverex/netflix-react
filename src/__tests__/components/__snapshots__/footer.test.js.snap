// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-alnijs-0 eORCAM"
>
  <p
    class="footer__Title-alnijs-4 iQvJpC"
  >
    Questions ? Contact Us.
  </p>
  <p
    class="footer__Break-alnijs-6 uTqSh"
  />
  <div
    class="footer__Row-alnijs-2 ZFMsH"
  >
    <div
      class="footer__Column-alnijs-1 jJuKdv"
    >
      <a
        class="footer__Link-alnijs-3 eDRVpg"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-alnijs-1 jJuKdv"
    >
      <a
        class="footer__Link-alnijs-3 eDRVpg"
        href="#"
      >
        Help Centre
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-alnijs-1 jJuKdv"
    >
      <a
        class="footer__Link-alnijs-3 eDRVpg"
      >
        Account
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
      >
        Redeem Gift Card
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
      >
        Privacy
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-alnijs-1 jJuKdv"
    >
      <a
        class="footer__Link-alnijs-3 eDRVpg"
      >
        Buy Gift Cards
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
      >
        Legal Notices
      </a>
      <a
        class="footer__Link-alnijs-3 eDRVpg"
      >
        Corporate Information
      </a>
    </div>
  </div>
  <p
    class="footer__Break-alnijs-6 uTqSh"
  />
  <p
    class="footer__Text-alnijs-5 hRoPqX"
  >
    Netflix United States
  </p>
</div>
`;
